src/Features/VisualBasic/Portable/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb

```diff
@@ -0,0 +1,175 @@
+            Dim castExpression = targetNode.Cast(conversionType, Nothing).WithAdditionalAnnotations(Simplifier.Annotation)
+            Dim newRoot = currentRoot.ReplaceNode(targetNode, castExpression)
+            Return newRoot
+        End Function
+
+        Protected Overrides Function TryGetTargetTypeInfo(document As Document, semanticModel As SemanticModel,
+                root As SyntaxNode, diagnosticId As String, spanNode As ExpressionSyntax,
+                cancellationToken As CancellationToken,
+                ByRef potentialConversionTypes As ImmutableArray(Of Tuple(Of ExpressionSyntax, ITypeSymbol))) As Boolean
+            potentialConversionTypes = ImmutableArray(Of Tuple(Of ExpressionSyntax, ITypeSymbol)).Empty
+
+            ' The error happens either on an assignement operation Or on an invocation expression.
+            ' If the error happens on assignment operation, "ConvertedType" Is different from the current "Type"
+            Dim mutablePotentialConversionTypes = ArrayBuilder(Of Tuple(Of ExpressionSyntax, ITypeSymbol)).GetInstance()
+            Select Case diagnosticId
+                Case BC30512, BC42016
+                    Dim argument = spanNode.GetAncestors(Of ArgumentSyntax).FirstOrDefault()
+                    If argument IsNot Nothing AndAlso argument.GetExpression.Equals(spanNode) Then
+                        ' spanNode is an argument expression
+                        Dim argumentList = argument.GetAncestorOrThis(Of ArgumentListSyntax)
+                        Dim invocationNode = argumentList.Parent
+
+                        mutablePotentialConversionTypes.AddRange(GetPotentialConversionTypes(semanticModel, root,
+                            argument, argumentList, invocationNode, cancellationToken))
+                    Else
+                        ' spanNode is a right expression in assignment operation
+                        Dim inferenceService = document.GetRequiredLanguageService(Of ITypeInferenceService)()
+                        Dim conversionType = inferenceService.InferType(semanticModel, spanNode, False, cancellationToken)
+                        mutablePotentialConversionTypes.Add(Tuple.Create(spanNode, conversionType))
+                    End If
+                Case BC30518, BC30519
+                    Dim invocationNode = spanNode.GetAncestors(Of ExpressionSyntax).FirstOrDefault(
+                        Function(node) Not node.ChildNodes.OfType(Of ArgumentListSyntax).IsEmpty())
+                    Dim argumentList = invocationNode.ChildNodes.OfType(Of ArgumentListSyntax).FirstOrDefault()
+
+                    ' spanArgument is null because the span is on the invocation identifier name according to BC30519
+                    mutablePotentialConversionTypes.AddRange(GetPotentialConversionTypes(semanticModel, root,
+                        spanArgument:=Nothing, argumentList, invocationNode, cancellationToken))
```

Comment:
consider extracting out these sections to helpers that can explain what cases they're working on and what they're doing.

Reply:
I am not sure I understand it, does it mean I can create a helper function that explain when `spanArgument` is null and when is not?